docs/04-operation-guides/troubleshooting/obsv-01-troubleshoot-prometheus-istio-server-crash-oom.md

```diff
@@ -2,79 +2,89 @@
 title: Prometheus Istio Server restarting or in crashback loop
 ---
 
-<!-- the entire content needs update: values files instead of configmaps -->
-
 ## Condition
 
 Prometheus Istio Server is restarting or in a crashback loop.
 
 ## Cause
 
-Prometheus Istio Server scrapes metrics from all envoy side cars. It might crash because of OOM when the cardinality of the Istio metrics increases too much. This usually happens when a high number of workloads perform a lot of communication to other workloads.
+*work in progress*
+
+Prometheus Istio Server scrapes metrics from all envoy side cars, which may lead to OOM issues.
+
+For example, this may happen when a high number of workloads perform a lot of communication to other workloads, or when workloads are created and deleted dynamically.
+
+Because Istio telemetry V2 currently doesn't support the concept of metric expiry, the cardinality of the Istio metrics increases too much. This causes the container to be killed because of OOM.
+
+There can be other causes for the Prometheus Istio Server to restart or crash, but the following istructions focus on fixing the OOM issue.
 
 ## Remedy
 
-1. Increase the memory for `prometheus-istio-server`:
+*work in progress*
+
+There are two ways to prevent the OOM issues:
 
-    ```bash
-    kubect edit deployment -n kyma monitoring-prometheus-istio-server
+- You can fix it on the Helm side.
+- You can configure your Kyma cluster with a specific values YAML file.
 
-    ```
+> **CAUTION:** Both procedures suggest dropping additional labels with `prometheus-istio.envoyStats.labeldropRegex`. As a side effect, graphs in Kiali will not work.
 
-    Increase the limits for memory:
+### Change the Helm configuration
 
-    ```yaml
-    resources:
-      limits:
-        cpu: 600m
-        memory: 2000Mi
-      requests:
-        cpu: 40m
-        memory: 200Mi
-    ```
+1. To increase the memory for `prometheus-istio-server`, run the following command:
 
-2. Drop labels for the Istio metrics.
+  ```bash
+  kubect edit deployment -n kyma monitoring-prometheus-istio-server
+  ```
 
-   Edit the values for `prometheus-istio server`:
+2. In your deployment resource, set the following limits for memory:
 
-   ```bash
-    kubectl edit configmap -n kyma-system monitoring-prometheus-istio-server
-    ```
+  ```yaml
+  resources:
+    limits:
+      cpu: 600m
+      memory: 2000Mi
```

Comment:
The way it's currently written, we don't really tell the users what exactly they are trying to accomplish with the suggested commands. I'm now getting the following picture:

There are two ways to solve the issue, both are independent from each other (?):
- increase memory limit
- drop the labels

And there are two alternative environments to do that:
- in Helm
- in the Kyma settings

Is that correct? If yes, I'll rewrite the doc to make that clearer.

Reply:
I notice you mention 6Gi, while the YAML file in the following section suggests 4Gi. Is there a reason to have different recommendations for the Helm settings and the Kyma settings? If not, I'll use 6Gi (or 4Gi if you prefer) for both.

Reply:
Feedback from @rakesh-garimella: 
The problem is we don't know the recommended value as this can be quite arbitrary so I would say 4Gi and 6Gi both are fine. But this then comes down to the customer how much he has free to give. So I think what you say is right if no scarcity of resources put max so 6Gi else increase in say steps of 1Gi